File: api/internal/domain/privacy_mode.go
package domain

// PrivacyCapabilities describes actual behavior rather than a marketing claim.
type PrivacyCapabilities struct {
	Mode             string                       `json:"mode"`
	NoEgress         bool                         `json:"noEgress"`
	RedactionEnabled bool                         `json:"redactionEnabled"`
	Capabilities     map[string]PrivacyCapability `json:"capabilities"`
}

// PrivacyCapability describes whether a potentially external capability is available.
type PrivacyCapability struct {
	Available bool   `json:"available"`
	Reason    string `json:"reason,omitempty"`
}

File: web/lib/privacy-mode.ts
import { api } from '@/lib/api';

export interface PrivacyCapabilities {
  mode: 'standard' | 'local_private';
  noEgress: boolean;
  redactionEnabled: boolean;
  capabilities: Record<
    string,
    {
      available: boolean;
      reason?: string;
    }
  >;
}

export const privacyModeApi = {
  getCapabilities: () => api.get<PrivacyCapabilities>('/api/public/privacy/capabilities'),
};
